File: shamelagpt-android/app/src/main/java/com/shamelagpt/android/domain/repository/AuthRepository.kt
package com.shamelagpt.android.domain.repository

import com.shamelagpt.android.data.remote.dto.AuthResponse
import com.shamelagpt.android.data.remote.dto.LoginRequest
import com.shamelagpt.android.data.remote.dto.SignupRequest
import com.shamelagpt.android.data.remote.dto.UpdateUserRequest
import com.shamelagpt.android.data.remote.dto.UserPreferencesRequest
import com.shamelagpt.android.data.remote.dto.UserResponse

interface AuthRepository {
    suspend fun signup(request: SignupRequest): Result<AuthResponse>
    suspend fun login(request: LoginRequest): Result<AuthResponse>
    suspend fun forgotPassword(email: String): Result<Unit>
    suspend fun googleSignIn(idToken: String): Result<AuthResponse>
    suspend fun refreshToken(refreshToken: String): Result<AuthResponse>
    suspend fun getCurrentUser(): Result<UserResponse>
    suspend fun updateCurrentUser(request: UpdateUserRequest): Result<UserResponse>
    suspend fun deleteCurrentUser(): Result<Unit>
    suspend fun verifyToken(): Result<Unit>
    suspend fun getPreferences(): Result<com.shamelagpt.android.data.remote.dto.UserPreferencesRequest>
    suspend fun setPreferences(request: UserPreferencesRequest): Result<Unit>
    fun logout()
    fun getToken(): String?
    fun isLoggedIn(): Boolean
}

File: shamelagpt-ios/shamelagpt/Domain/Repositories/AuthRepository.swift
//
//  AuthRepository.swift
//  ShamelaGPT
//
//  Created by Codex on 05/12/2025.
//

import Foundation

protocol AuthRepository {
    func signup(request: SignupRequest) async throws -> AuthResponse
    func login(request: LoginRequest) async throws -> AuthResponse
    func forgotPassword(email: String) async throws
    func googleSignIn(request: GoogleSignInRequest) async throws -> AuthResponse
    func refreshToken(request: RefreshTokenRequest) async throws -> AuthResponse
    func getCurrentUser() async throws -> UserResponse
    func updateCurrentUser(request: UpdateUserRequest) async throws -> UserResponse
    func deleteCurrentUser() async throws
    func verifyToken() async throws
    func getPreferences() async throws -> UserPreferencesRequest
    func setPreferences(_ request: UserPreferencesRequest) async throws
    func logout()
    func token() -> String?
    func isLoggedIn() -> Bool
}
